src/pages/table/table-editable.index.jsx

```diff
@@ -0,0 +1,191 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+import { createRoot } from 'react-dom/client';
+import React, { useRef, useState } from 'react';
+import { useCollection } from '@cloudscape-design/collection-hooks';
+import { Button, Pagination, Table, TextFilter } from '@cloudscape-design/components';
+import { distributionEditLabels, paginationLabels } from '../../common/labels';
+import { useLocalStorage } from '../../common/localStorage';
+import { getFilterCounterText } from '../../common/tableCounterStrings';
+import '../../styles/base.scss';
+import {
+  CustomAppLayout,
+  Navigation,
+  Notifications,
+  TableEmptyState,
+  TableNoMatchState,
+} from '../commons/common-components';
+import DataProvider from '../commons/data-provider';
+import { useColumnWidths } from '../commons/use-column-widths';
+import { Breadcrumbs, FullPageHeader, ToolsContent } from './common-components';
+import {
+  EDITABLE_PREFERENCES,
+  EDITABLE_COLUMN_DEFINITIONS,
+  Preferences,
+  serverSideErrorsStore,
+  domainNameRegex,
+  INVALID_DOMAIN_MESSAGE,
+} from './table-config';
+
+const withSideEffect =
+  (fn, sideEffect) =>
+  (...args) => {
+    sideEffect(...args);
+    return fn(...args);
+  };
+
+const fakeDataFetch = delay => new Promise(resolve => setTimeout(() => resolve(), delay));
+
+function TableContent({ loadHelpPanelContent, distributions }) {
+  const [loading, setLoading] = useState(false);
+  const [tableItems, setTableItems] = useState(distributions);
+  const [columnDefinitions, saveWidths] = useColumnWidths('React-EditableTable-Widths', EDITABLE_COLUMN_DEFINITIONS);
+  const [preferences, setPreferences] = useLocalStorage(
+    'React-EditableDistributionsTable-Preferences',
+    EDITABLE_PREFERENCES
+  );
+  const [itemsSnap, setItemsSnap] = useState(null);
```

Comment:
What does "snap" mean in this context?

In general, it might be helpful to add a few comments in place where we do inline-editing specific things. I realize this is not something we do with other demos, but as these are getting more complex it would be helpful.

Reply:
Updated the name to be more descriptive